homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,282 @@
+"""
+Support for Honeywell Lyric thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/lyric
+"""
+import logging
+from typing import List, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    ATTR_TIME,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+import homeassistant.helpers.config_validation as cv
+from . import LyricDeviceEntity
+from .const import DATA_LYRIC_CLIENT, DATA_LYRIC_DEVICES, DOMAIN, SERVICE_HOLD_TIME
+
+_LOGGER = logging.getLogger(__name__)
+
+PRESET_NO_HOLD = "NoHold"
+PRESET_HOLD_UNTIL = "HoldUntil"
+PRESET_PERMANENT_HOLD = "PermanentHold"
+PRESET_TEMPORARY_HOLD = "TemporaryHold"
+PRESET_VACATION_HOLD = "VacationHold"
+
+SUPPORT_FLAGS = (
+    SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE_RANGE
+)
+
+HOLD_PERIOD_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.comp_entity_ids,
+        vol.Required(ATTR_TIME): cv.string,
+    }
+)
+
+LYRIC_HVAC_MODES = {
+    HVAC_MODE_OFF: "OFF",
+    HVAC_MODE_HEAT: "HEAT",
+    HVAC_MODE_COOL: "COOL",
+    HVAC_MODE_HEAT_COOL: "HEAT_COOL",
+}
+
+HVAC_MODES = {
+    "OFF": HVAC_MODE_OFF,
+    "HEAT": HVAC_MODE_HEAT,
+    "COOL": HVAC_MODE_COOL,
+    "HEAT_COOL": HVAC_MODE_HEAT_COOL,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Lyric thermostat based on a config entry."""
+    lyric = hass.data[DOMAIN][DATA_LYRIC_CLIENT]
+
+    try:
+        devices = lyric.devices()
```

Comment:
if this is doing I/O, it should do an `await`. or you need to run it in the executor. You're inside an async context.

Reply:
```suggestion
        devices = await hass.async_add_executor_job(lyric.devices)
```